src/initialize.F90

```diff
@@ -1120,4 +1133,57 @@ recursive subroutine allocate_offsets(univ_list, counts, found)
 
   end subroutine allocate_offsets
 
+!===============================================================================
+! LOOKUP_MATERIAL_TEMPERATURES If any cells have undefined temperatures, try to
+! find their temperatures from material data.
+!===============================================================================
+
+  subroutine lookup_material_temperatures()
+    integer :: i, j, k
+    real(8) :: min_temp
+    logical :: warning_given
+
+    warning_given = .false.
+    do i = 1, n_cells
+      ! Ignore non-normal cells and cells with defined temperature.
+      if (cells(i) % type /= CELL_NORMAL) cycle
+      if (cells(i) % sqrtkT(1) /= ERROR_REAL) cycle
+
+      ! Set the number of temperatures equal to the number of materials.
+      deallocate(cells(i) % sqrtkT)
+      allocate(cells(i) % sqrtkT(size(cells(i) % material)))
+
+      ! Check each of the cell materials for temperature data.
+      do j = 1, size(cells(i) % material)
+        ! Arbitrarily set void regions to 0K.
+        if (cells(i) % material(j) == MATERIAL_VOID) then
+          cells(i) % sqrtkT(j) = ZERO
+          cycle
+        end if
+
+        associate (mat => materials(cells(i) % material(j)))
+          ! Find the temperature of the coldest nuclide.
+          min_temp = nuclides(mat % nuclide(1)) % kT
+          do k = 2, mat % n_nuclides
+            ! Warn the user if the nuclides don't have identical temperatues.
+            if (nuclides(mat % nuclide(k)) % kT /= min_temp &
+                 .and. .not. warning_given) then
```

Comment:
We don't need to spit out this warning if a user isn't using multipole at all so add `.and. multipole_active` into that `if` statement.